### ISSUE TITLE
Feature Request: Weather

### ISSUE THREAD

**Issue by Jump-Suit** [OpenRCT2/OpenLoco#131]
One thing that people tend to miss is a game like this is, weather. Where is the weather? Roller Coaster Tycoon had weather yet Chris Sawyer's Locomotion doesn't? How does that add up? Well in any case, to expand upon it. I am thinking adding not just the rain and what not like Chri did back in RCT. But add Tornados or something that is capable of doing physical damage to terrain and vehicles, tracks etc. This could add a whole new aspect to the game and make it more a challenge of being weary of the weather and moving your assets to a safe area. 
Thoughts?

**Comment by auge8472** [comment]
In RCT weather has a direct influence to the gameplay. In a transport game it hasn't. So why should it be there? What will it add to the gameplay? Will you alter the timetables of your vehicles depending from rainy, snowy or dry weather?

*IMHO* it is an unnecessary "non"-feature.

**Comment by ItsTheKais** [comment]
I feel like being made to stop what you're doing and pick up a bunch of your vehicles to dodge an arbitrary punishment wouldn't make for fun gameplay.

Rain or snow could still have passive effects, though, like they do in real life. Road vehicles have to drive slower. Aircraft get delays and have to sit at airports for longer periods between flights. Trains lose traction and spend more time braking and accelerating than driving at full speed. Uncovered cargo like paper or iron ore might be damaged and lose value, and stations get harsher rating penalties if cargo is left sitting in the rain.

It's probably not going to be added anytime remotely soon, and it definitely needs toggles if/when it happens, but I don't think it's entirely a "non-feature."

**Comment by jgottula** [comment]
Locomotion does have *climate* simulation (i.e. the longer-term seasonal sort of stuff, rather than short-term weather stuff). So depending on the particular scenario, the weather 

> In RCT weather has a direct influence to the gameplay.

Right. In RCT, weather had obvious and direct impacts on the gameplay:

- Rain: peeps will largely not want to go on outdoor/uncovered rides
- Rain: peeps will buy your cruelly overpriced $20 umbrellas because they don't want to get wet
- Hot weather: peeps will want to buy ice cream and lemonade and whatnot (I think? not 100% sure on that one I guess)

Locomotion also kind of operates on a different scale, both in terms of the size/scale of the world and the things in it (big difference compared to RCT), and in terms of the passage of time (both the real-time scale on which things move, and the background time scale of the calendar; the difference from RCT here isn't as big as the scale difference, but it still does feel rather different).

> OP: [adding rain]

Rain might be fine as a cosmetic feature, I suppose... but a bit pointless and somewhat misleading unless it had actual gameplay effects. (Which I'll address below.)

> But add Tornados or something that is capable of doing physical damage to terrain and vehicles, tracks etc. This could add a whole new aspect to the game and make it more a challenge of being weary of the weather and moving your assets to a safe area.

I agree with @ItsTheKais's sentiment on this: it would primarily just be tedious and not particularly fun.

Also it would be a significantly large change to the overall gameplay of Locomotion; and that's not really the primary focus of this project. I don't necessarily mean to speak for the OpenRCT2/OpenLoco team as a whole, but I think our main goals are more about improving these games as they already are, rather than making substantial changes to their core gameplay experience. (Fixing bugs, improving limitations, cross-platform support, better graphics performance, additional minor features that don't hugely change the way the game is fundamentally played, etc.)

> Rain or snow could still have passive effects, though, like they do in real life. Road vehicles have to drive slower. Aircraft get delays and have to sit at airports for longer periods between flights. Trains lose traction and spend more time braking and accelerating than driving at full speed. Uncovered cargo like paper or iron ore might be damaged and lose value, and stations get harsher rating penalties if cargo is left sitting in the rain.

This *does* sound somewhat interesting, and it could potentially make for an additional layer of gameplay and strategy considerations and so forth.

But like I mentioned above previously, this would be a pretty substantial change to the gameplay of Locomotion, and that's really not our primary project goal. Also, implementing this would be a *lot* of work, for something that I suspect many people would ultimately just consider to be an annoyance and leave disabled. :confused: 

So I actually do agree that it isn't entirely a "non-feature". But at the same time, I highly doubt that this is the sort of thing that would ever likely get implemented at all (let alone be a high priority).

----

Oh, and to add to the whole "this would be a lot of work" point I've been making: this would *most definitely* require us to have implemented the new/extended save format. Which means that (a) this is a feature that wouldn't be possible for quite a while even in principle; and (b) saved games wouldn't be backwards-compatible with vanilla Locomotion (like, at all).

**Comment by jgottula** [comment]
> One thing that people tend to miss is a game like this is, weather. Where is the weather?

Well, something you might be overlooking here is that Locomotion *does* have *climate* simulation (i.e. the longer-term seasonal sort of stuff, rather than short-term weather stuff). So scenarios do actually choose the particular type of climate they want to have:
![scenarioeditorclimate](https://user-images.githubusercontent.com/822207/36457847-e9d2f844-1660-11e8-8ee0-2ef46c22feab.png)

And then depending on that climate type selection and the geography of the map, the climate *does* actually have some real effects on things.

The most obvious example I can think of off the top of my head is that grain farms, livestock farms, and vineyards whose fields are at high altitude (above the snow line) will have their production fall down to essentially zero for months during the winter while those fields are covered by snow (assuming that the scenario has a climate selected that includes snow). And that can have a pretty significant gameplay impact.

(If you don't believe me, go and play the medium difficulty "Vache and Vineyards" scenario, where pretty much the only available industries are farms and vineyards, and most of them are above the snow line; it really makes that scenario more difficult than it'd be with no climate effects.)

----

> In RCT weather has a direct influence to the gameplay.

Right. In RCT, weather had direct and obvious impacts on the gameplay:

- Rain: peeps will largely not want to go on outdoor/uncovered rides
- Rain: peeps will buy your cruelly overpriced $20 umbrellas because they don't want to get wet
- Hot weather: peeps will want to buy ice cream and lemonade and whatnot (I think? not 100% sure on that one I guess)

Locomotion also kind of operates on a different scale, both in terms of the size/scale of the world and the things in it (big difference compared to RCT), and in terms of the passage of time (both the real-time scale on which things move, and the background time scale of the calendar; the difference from RCT here isn't as big as the scale difference, but it still does feel rather different).

----

> OP: [adding rain]

Rain might be fine as a cosmetic feature, I suppose... but a bit pointless and somewhat misleading unless it had actual gameplay effects. (Which I'll address later below.)

----

> But add Tornados or something that is capable of doing physical damage to terrain and vehicles, tracks etc. This could add a whole new aspect to the game and make it more a challenge of being weary of the weather and moving your assets to a safe area.

I agree with @ItsTheKais's sentiment on this: it would primarily just be tedious and not particularly fun.

Also it would be a significantly large change to the overall gameplay of Locomotion; and that's not really the primary focus of this project. I don't necessarily mean to speak for the OpenRCT2/OpenLoco team as a whole, but I think our main goals are more about improving these games as they already are, rather than making substantial changes to their core gameplay experience. (Fixing bugs, improving limitations, cross-platform support, better graphics performance, additional minor features that don't hugely change the way the game is fundamentally played, etc.)

----

> Rain or snow could still have passive effects, though, like they do in real life. Road vehicles have to drive slower. Aircraft get delays and have to sit at airports for longer periods between flights. Trains lose traction and spend more time braking and accelerating than driving at full speed. Uncovered cargo like paper or iron ore might be damaged and lose value, and stations get harsher rating penalties if cargo is left sitting in the rain.

This *does* sound somewhat interesting, and it could potentially make for an additional layer of gameplay and strategy considerations and so forth.

But like I mentioned above previously, this would be a pretty substantial change to the gameplay of Locomotion, and that's really not our primary project goal. Also, implementing this would be a *lot* of work, for something that I suspect many people would ultimately just consider to be an annoyance and leave disabled. :confused: 

So I actually do agree that it isn't entirely a "non-feature". But at the same time, I highly doubt that this is the sort of thing that would ever likely get implemented at all (let alone be a high priority).

----

Oh, and to add to the whole "this would be a lot of work" point I've been making here: this would *most definitely* require us to have implemented the new/extended save format. Which means that (a) this is a feature that wouldn't be possible for quite a while even in principle; and (b) saved games wouldn't be backwards-compatible with vanilla Locomotion (like, at all).

**Comment by auge8472** [comment]
> Rain or snow could still have passive effects, though, like they do in real life. Road vehicles have to drive slower. Aircraft get delays and have to sit at airports for longer periods between flights. Trains lose traction and spend more time braking and accelerating than driving at full speed. Uncovered cargo like paper or iron ore might be damaged and lose value, and stations get harsher rating penalties if cargo is left sitting in the rain.

You are correct. If we had weather in the game all this could happen. But again: What's the influence to the gameplay? What **you** as a player would change?

The additional way and time for breaking or accelerating a vehicle would be an automatism of the vehicle. Also the delay of aircrafts would be an automatism. Cargos that could lose value should be transported in appropriate carriages/vehicles. Paper is actually transported on flat cars, whith weather it would be transported in a goods van instead. This is IMHO a task for vehicle (set) creators and is the only influence to the gameplay (I can imagine ad hoc).

To make the game behaviour a bit more erratic (varying travel time because of accelerating and breaking slower in rainy or snowy weather) as the price for a different selection of vehicles is *IMHO* not a good idea.
